src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -362,13 +362,19 @@ function reducer(draft, action) {
         schema.defaultValue = null;
       }
 
+      // TODO: This could break if a reusable definition is used multiple times in a form
+      const step = draft.formSteps.filter(
+        step => step.formDefinition === formDefinition || step._generatedId === formDefinition
+      )[0];
```

Comment:
* you can/should use `Array.find` for this rather than `.filter(...)[0]`
* looks a bit like a candidate for https://github.com/open-formulieren/open-forms/blob/master/src/openforms/js/components/admin/form_design/StepSelection.js#L35 or rather split out the matching/filter functionality into a utility function that's used here and in the hook? I'm looking to "contain" the implementation details of the private `_generatedId` vs. the URL thing